conference/src/main/java/com/kyriba/conference/management/api/HallController.java

```diff
@@ -43,45 +44,45 @@ public HallController(HallService hallService)
 
   @ApiOperation(value = "Get a conference hall if exists")
   @GetMapping(value = "/{id}", produces = APPLICATION_JSON_UTF8_VALUE)
-  HallResponse findHall(@Valid @ApiParam(value = "Hall identity", required = true) @PathVariable Long id)
+  ResponseEntity<HallResponse> findHall(
+      @Valid @ApiParam(value = "Hall identity", required = true) @PathVariable Long id)
   {
-    return hallService.find(id)
-        .map(HallResponse::new)
-        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Hall not found"));
+    return ResponseEntity.of(hallService.findHall(id));
   }
 
 
   @ApiOperation(value = "Show all conference halls", responseContainer = "List")
   @GetMapping(produces = APPLICATION_JSON_UTF8_VALUE)
   List<HallResponse> findAllHalls()
   {
-    return hallService.findAll().stream()
-        .map(HallResponse::new)
-        .collect(Collectors.toList());
+    return hallService.findAllHalls();
   }
 
 
   @ApiOperation(value = "Create conference hall")
   @PostMapping(produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
-  Long createHall(@Valid @ApiParam(value = "Hall creation object", required = true) @RequestBody HallRequest hall)
+  Long createHall(
+      @Valid @ApiParam(value = "Hall creation object", required = true) @RequestBody HallRequest hallRequest)
   {
-    return hallService.createHall(hall).getId();
+    return hallService.createHall(hallRequest);
   }
 
 
   @ApiOperation(value = "Change conference hall parameters")
   @PutMapping(value = "/{id}", produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
-  void updateHall(@Valid @ApiParam(value = "Hall identity", required = true) @PathVariable String id,
-                  @Valid @ApiParam(value = "Hall change object", required = true) @RequestBody HallRequest hall)
+  void updateHall(@Valid @ApiParam(value = "Hall identity", required = true) @PathVariable Long id,
+                  @Valid @ApiParam(value = "Hall change object", required = true) @RequestBody HallRequest hallRequest) throws
+      EntityNotFound
   {
-    hallService.updateHall(Long.valueOf(id), hall);
+    hallService.updateHall(id, hallRequest);
```

Comment:
What about the case when the hall with this id doesn't exist?

Reply:
done